tour_model_eval/viz_bin_cluster.py

```diff
@@ -15,51 +15,40 @@
 import branca.colormap as cm
 
 
-def bins_map(bins, bins_list, trips):
+def bins_map(bins, sel_bin_indices, trips):
     # Plot all the bin trips on the map, use different colors for different bins.
     # Each color represents one bin based on index
     # Choose the first start point from the trips to locate the map for convenience
 
     # The function takes three parameters:
     # - bins: bins list above cutoff after running pipeline.remove_noise
-    # - bins_list: the list of indices of selected bins
+    # - sel_bin_indices: the list of indices of selected bins (set to None for plotting all bins)
     # - trips: the trips after running pipeline.read_data
 
     # In order to plot trips correctly, map index should be len(bins)+1,
     # so that the color steps length is equal to the bins number. For bin index which is 0,
     # will be plotted in color which is in index 1.
-    color_map = cm.linear.Set1_07.to_step(len(bins)+1, index=[i for i in range(len(bins)+1)])
+    color_map = cm.linear.Set1_07.to_step(len(bins), index=[i for i in range(len(bins)+1)])
```

Comment:
I am not going to force you to change this, but I want to point out that you can in fact get colors out of a colormap object. In fact, there are two ways to do so:

First, we create the colormap

```
>>> import branca.colormap as cm
>>> color_map = cm.linear.Set1_07.to_step(10, index=range(10))
>>> color_map
<branca.colormap.StepColormap object at 0x7f91c8356fd0>
```

Option 1: create a list through iteration.

```
>>> [color_map(x) for x in range(10)]
['#e41a1c', '#e41a1c', '#626591', '#429781', '#609760', '#984ea3', '#e67328', '#ffbf19', '#e9d530', '#a65628']
```

Option 2: access the `colors` property directly. This is not a documented interface, so option 1 is better.

```
>>> color_map.colors
[(0.8941176470588236, 0.10196078431372549, 0.10980392156862745, 1.0), (0.4418300653594771, 0.36339869281045756, 0.5176470588235295, 1.0), (0.24444444444444446, 0.5581699346405229, 0.5777777777777778, 1.0), (0.3019607843137256, 0.6862745098039215, 0.2901960784313727, 1.0), (0.49803921568627446, 0.4326797385620916, 0.522875816993464, 1.0), (0.7307189542483661, 0.3699346405228759, 0.4261437908496731, 1.0), (0.9999999999999998, 0.4980392156862744, 4.2580318356211883e-16, 1.0), (1.0, 0.8326797385620914, 0.13333333333333333, 1.0), (0.8836601307189542, 0.7790849673202613, 0.18562091503267975, 1.0), (0.6509803921568628, 0.33725490196078434, 0.1568627450980392, 1.0)]

>>> import matplotlib
>>> matplotlib.colors.to_hex(color_map.colors[0])
'#e41a1c'

>>> [matplotlib.colors.to_hex(c) for c in color_map.colors]
['#e41a1c', '#715d84', '#3e8e93', '#4daf4a', '#7f6e85', '#ba5e6d', '#ff7f00', '#ffd422', '#e1c72f', '#a65628']
```

Reply:
Ah, there is somewhere online pointed out if index is given, the colormap will follow the index. So, no matter what I put on, as long as the index is correct, the colormap will be fine. I think it's not a big deal there.

Reply:
As I said, I am not going to force you to fix this. But I didn't believe that you can't get a color list out of the colormap, so I did that little thought experiment.